datastore/importers/nmdb-import-ipconfig/Parser.cpp

```diff
@@ -36,21 +36,31 @@ Parser::Parser() : Parser::base_type(start)
     // Skip garbage before
     *(!qi::lit("Windows IP Configuration") >> -qi::omit[+token] >> qi::eol) >>
     (qi::lit("Windows IP Configuration") > +qi::eol >>
-     +(-compartmentHeader >>
-       +( ("Host Name" >> dots > fqdn > qi::eol)
-            [(pnx::bind(&Parser::addDevInfo, this, qi::_1))]
-        | ("Primary Dns Suffix" >> dots > -fqdn > qi::eol)
-        | ignoredLine
-       ) >> qi::eol >>
-       *(adapter)
-      )
-    ) [(qi::_val = pnx::bind(&Parser::getData, this))] >>
+     (
+      +adapter |
+      +(
```

Comment:
Since both of these are at least one, you probably want to pull the `+` to the outside.  I am assuming you mean one or more of either and not one or more of only one section.

Reply:
The thought here is that they would be mutually exclusive. With a bare ipconfig, it would only be adapters, while one with added options would then include optional compartments or the host data.

Reply:
Hmmm, that's a fair thought process.  Hopefully they are mutually exclusive.  I believe the reasoning for the `/allcompartments` flag was was abandoned right after it was introduced (i.e., I haven't seen it even documented since Windows 7).  However, who knows what they intend to do since they haven't removed it yet.